db.go

```diff
@@ -521,7 +519,7 @@ func (db *DB) reload() (err error) {
 		// See if we already have the block in memory or open it otherwise.
 		b, ok := db.getBlock(meta.ULID)
 		if !ok {
-			b, err = OpenBlock(dir, db.chunkPool)
+			b, err = OpenBlock(dir, nil)
```

Comment:
are you sure this doesn't increase allocations?

if you pass the same pool to `OpenBlock()` it should help?

Reply:
I could have been a bit more clear, this issue is not an allocation issue, but just an optimization to avoid the overhead of using a `sync.Pool`. Since we never put anything into the pool except during a compaction, the pool is almost always empty so we have to allocate a new chunk anyway.

I could look into trying to add chunks back to the pool, but it is hard to know exactly when we would be able to. Perhaps when a querier is closed?

Reply:
after a compactoin we put back in the Pool and right after that we reload the db which calls OpenBlock so if these share the same pool should reduce allocations, no?

I think a dedicated benchmark would answer the question.

Reply:
ping @csmarchbanks

Reply:
I will do some work looking at testing with a dedicated pool today.

Having the pool will likely help allocations a bit right after a compaction, either until the first garbage collection (see https://github.com/golang/go/issues/22950), or enough queries have run such that the pool is drained. But in reality, the only allocation that will be saved is the struct (the buffer is replaced after a Get), so it would be a tiny amount of allocations saved.

Reply:
thanks for the update , if a bench shows the same than  good for merge.

I like deleting code :stuck_out_tongue_winking_eye:

Reply:
I tried using a combined pool, and it didn't seem to help with allocations at all. I would like to try a few more things, to make sure I am actually testing what I think I am. I will report back soon!

Reply:
I did some more benchmarking, and it really seems like the overhead of syncPool is more expensive than any caching gains for the small amount of time around a compaction. I went as far as having the querier return chunks to the pool when it was done with them, but it was still slower.

It looks like the syncPool was originally implemented to reduce spikes when compacting, and that path will still use a pool (https://github.com/prometheus/tsdb/pull/118).

Since it's only a couple percent gain for retrieving and iterating through a chunk, I understand if it feels a bit risky to merge though!

Reply:
Can this code be added to any of the test files so I can have a look?

Reply:
Also, here's the result of current master, compared to adding a put to the Pool in the Reader:
```
benchcmp original.txt with_pool.txt
benchmark                                                          old ns/op      new ns/op      delta
BenchmarkPersistedQueries/series=10,samplesPerSeries=1000-8        1100883        1112310        +1.04%
BenchmarkPersistedQueries/series=10,samplesPerSeries=10000-8       10623663       10533229       -0.85%
BenchmarkPersistedQueries/series=10,samplesPerSeries=100000-8      105319888      106529656      +1.15%
BenchmarkPersistedQueries/series=100,samplesPerSeries=1000-8       10772258       10645391       -1.18%
BenchmarkPersistedQueries/series=100,samplesPerSeries=10000-8      105563054      105657945      +0.09%
BenchmarkPersistedQueries/series=100,samplesPerSeries=100000-8     1128303811     1103207237     -2.22%

benchmark                                                          old allocs     new allocs     delta
BenchmarkPersistedQueries/series=10,samplesPerSeries=1000-8        520            358            -31.15%
BenchmarkPersistedQueries/series=10,samplesPerSeries=10000-8       3563           2037           -42.83%
BenchmarkPersistedQueries/series=10,samplesPerSeries=100000-8      34092          18999          -44.27%
BenchmarkPersistedQueries/series=100,samplesPerSeries=1000-8       4587           2806           -38.83%
BenchmarkPersistedQueries/series=100,samplesPerSeries=10000-8      34997          18256          -47.84%
BenchmarkPersistedQueries/series=100,samplesPerSeries=100000-8     340262         179192         -47.34%

benchmark                                                          old bytes     new bytes     delta
BenchmarkPersistedQueries/series=10,samplesPerSeries=1000-8        68170         64940         -4.74%
BenchmarkPersistedQueries/series=10,samplesPerSeries=10000-8       212348        181887        -14.34%
BenchmarkPersistedQueries/series=10,samplesPerSeries=100000-8      1713962       1415349       -17.42%
BenchmarkPersistedQueries/series=100,samplesPerSeries=1000-8       305244        269624        -11.67%
BenchmarkPersistedQueries/series=100,samplesPerSeries=10000-8      1662664       1330012       -20.01%
BenchmarkPersistedQueries/series=100,samplesPerSeries=100000-8     15942680      12854432      -19.37%
```